### DISCUSSION TITLE
Protocol | Nov 25, 2022 | ForDem PM Weekly

### DISCUSSION THREAD

**Discussion by tcf-admin** [tcfev/forDem#136] [Announcements]
Attendees: @armantorkzaban  @lukas-h @zanzendegi 

Notes
	•	@Kobin-ts could not join. Internet problems
	•	@lukas-h Lukas and @zanzendegi met each other for the first time, they introduced themselves to each other and (background, interests etc)

Topic Notes
**Feature Engineering**
Needs improvement over GithubLukas talks about Scalabs/Documentation standards*proposes to adopt the same for forDemTemplates were discussed https://github.com/scalabs/documentation

**Quality control**
Done via Code reviews @lukas-h offers daily code review & sharing technical opinion

**Architecture of the team**
https://github.com/tcfev/forDem/blob/main/.images/organisation.md 
@lukas-h : 
- Decision making is board's
- no need for a CEO
- Teams with clear objectives
- Documentations dictate what needs to be done
- Add prioritisations to feature requests
- Once sprints are set release candidates are clear
- Decisions are made in the board
- Majority of present members of the board
-  - Follows the TCF constitution
- After the decision is made revisions can be proposed for the next iteration/release 
@zanzendegi:
- Agrees with teams and clear objectives
- For release we should have clear candidates
- Keep it simple and flexible

**Architecture of the platform**
We did not manage to review this.

**Bug Report**
Steps to reproduce the bug.https://github.com/scalabs/documentation/blob/main/.github/ISSUE_TEMPLATE/bug_report.md (a template)
Adopt Scalabs Templates/Docs.
We all agreed to go for it.

**Periodic Meeting**
Min. 2 times a week Friday's meeting is already set.

**Branching**
PR merges for now can take place before review.*We need a clear comprise between speed an quality

**NewNode**
We did not manage to review this.  

**Action items**
	•	Find another meeting time during the week - invite Lukas
	•	Add @lukas-h handle to conversations and PRs - Lukas will **reply** **daily**
	•	In the beginning we will create a protected ‘release’ branch. @lukas-h @zanzendegi will propose a strategy
	•	Clear the **todo** and **inProgress** tasks in the kanban board of **app**, **backend** and **website** till the end of the month

**Suggestions**  follow

**Comment by zanzendegi** [comment]
Suggestion:

- In the next meet all participants share a doc (preferably Markdown) and collaborate in the note taking and highlighting of the items.
- Then the project director will transpile the notes into the project `Theme`s and `Epic`s. (@armantorkzaban , maybe you?)
- The UX designers (and everyone) then create `User Story`s and verify if we are on the path.
- Then other members derive the `Task`s from the Themes/Epics. And assign them.


More info on Theme, Epic, Task and User Story here: https://www.visual-paradigm.com/scrum/theme-epic-user-story-task/

**Comment by zanzendegi** [comment]
Markdown collaboration platform:

https://www.gitbook.com/pricing

**Comment by tcf-admin** [comment]
Moved to https://github.com/tcfev/forDem/blob/main/.assets/.protocols/protocols.md
@zanzendegi  @Kobin-ts @lukas-h  you are all invited to this collaborative .md editing space: https://hackmd.io/team/forDem

**Comment by lukas-h** [comment]
> • In the beginning we will create a protected ‘release’ branch. @lukas-h @zanzendegi will propose a strategy

Hi @zanzendegi, as a release strategy we could follow the standard at my company, as seen here:

https://github.com/scalabs/documentation/blob/main/guides/branches-releases-versioning.md#software-release-process

We had the idea to leave the `main` branches of the repos "open" to direct commits and merging, so the only thing to do is:
- [x] add a `releases` branch
- [x] add a branch protection rule, that allows merging only with a code review of another maintainer in the organization

**Comment by lukas-h** [comment]
> Clear the **todo** and **inProgress** tasks in the kanban board of **app**, **backend** and **website** till the end of the month

@armantorkzaban & everyone else.

We need to find time re-init and fill the project boards. I'm open to proposals.

**Reply by lukas-h** [reply]
We have started this effort with A NEW PROJECT BOARD: https://github.com/orgs/tcfev/projects/7

**Comment by lukas-h** [comment]
Another thing that we talked about during the meeting was, to establish **issue templates**. This can really help to create a certain standard of feature engineering and clearer documentation of the tasks at hand.

We at @scalabs use the following templates for **feature requests**, **bug reports**, and **questions**.
https://github.com/scalabs/github-workflows/tree/main/.github/ISSUE_TEMPLATE

We should add them to all the repos associated with the project

**Reply by lukas-h** [reply]
Issue templates are now set as a standard for all **ForDem** repositories

**Comment by lukas-h** [comment]
I think, it makes more sense to have this as a discussion rather than an issue, so this does not interfere with our project management

**Reply by lukas-h** [reply]
I went ahead and converted it
